### PR TITLE
[SEDONA-197] Added ST_ZMax and ST_ZMin to Apache Sedona

### PR DESCRIPTION
**Pull request by Varsha-R**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-197. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
Added ST_ZMax and ST_ZMin functions to SQL

## How was this patch tested?
Through unit tests written in Scala

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation.

### CODE REVIEW

File: docs/api/sql/Function.md [hunk 1]
```diff
@@ -1445,3 +1445,33 @@ SELECT ST_Z(ST_POINT(0.0 25.0 11.0))
...
 
 Output: `11.0`
+
+## ST_ZMax
+
+Introduction: Returns Z maxima of the given geometry or null if there is no Z coordinate.
+
+Format: `ST_ZMax(geom: geometry)`
+
+Since: `v1.2.1`
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:1455]: the version number should be 1.3.1

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Please add the following:

1. DataFrameAPI for Spark, with tests: https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala
2. Corresponding function in Flink, with tests: https://github.com/apache/incubator-sedona/blob/master/flink/src/main/java/org/apache/sedona/flink/Catalog.java#L21
3.

**Comment by Varsha-R** [comment]
> Please add the following:
> 
> 1. DataFrameAPI for Spark, with tests: https://github.com/apache/incubator-sedona/blob/master/sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/st_functions.scala
> 2. Corresponding function in Flink, with tests: https://github.com/apache/incubator-sedona/blob/master/flink/src/main/java/org/apache/sedona/flink/Catalog.java#L21

Addressed these comments in [this commit](https://github.com/apache/incubator-sedona/pull/717/commits/b9fffd6ea2830350128141c548372613b658fe00#diff-42591a191598418405b17af4a87d795eb556c69b5233d699e4c106230c0e99ef)

File: docs/api/sql/Function.md (hunk 1 above)
Comment (Varsha-R) [reply docs/api/sql/Function.md]: Updated it

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Looks good to me. One last thing: please add the DataFrame style API to Python as well -> https://github.com/apache/incubator-sedona/blob/master/python/sedona/sql/st_functions.py

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**